drivers/lis2dh12/lis2dh12.c

```diff
@@ -524,11 +537,13 @@ int lis2dh12_restart_fifo(const lis2dh12_t *dev)
     fifo_reg.reg = _read(dev, REG_FIFO_CTRL_REG);
 
     uint8_t fifo_mode_old = fifo_reg.bit.FM;
+    /* cppcheck-suppress unreadVariable; (reason: union type is used for read) */
     fifo_reg.bit.FM = LIS2DH12_FIFO_MODE_BYPASS;
 
     /* switch to Bypass mode */
     _write(dev, REG_FIFO_CTRL_REG, fifo_reg.reg);
 
+    /* cppcheck-suppress redundantAssignment; (reason: union type is used for read) */
```

Comment:
@maribu Here I don't know how to solve the `cppcheck` error.

If I use `cppcheck-suppress redundantAssignment`, `cppcheck` is complaining about unread variable. If I replace `cppcheck-suppress redundantAssignment` by `cppcheck-suppress unreadVariable`, it is complaining about redundant assignment.  

How can this be solved?

Reply:
Solved. I just have to use multiple `cppcheck-suppress` lines for all different warning that are given here.